dialer.go

```diff
@@ -116,6 +132,16 @@ func NewDialer(ctx context.Context, opts ...Option) (*Dialer, error) {
 		cfg.rsaKey = key
 	}
 
+	// If no token source is configured, use ADC's token source.
+	ts := cfg.tokenSource
+	if ts == nil {
+		var err error
+		ts, err = google.DefaultTokenSource(ctx, "https://www.googleapis.com/auth/cloud-platform")
```

Comment:
you could use `CloudPlatformScope` from options.go

Reply:
Fixed.